ui/packages/lib/src/components/nav_drawer/NavDrawer.js

```diff
@@ -12,38 +12,42 @@ import {
   EuiSpacer,
   EuiTreeView
 } from "@elastic/eui";
-import ApplicationsContext from "../../providers/application/context";
-import { CurrentProjectContext } from "../../providers/project";
 import { useToggle } from "../../hooks";
-import { navigate } from "@reach/router";
-import { get, slugify } from "../../utils";
+import { slugify } from "../../utils";
 import urlJoin from "proper-url-join";
+import { useNavigate } from "react-router-dom";
+import { ApplicationsContext, ProjectsContext } from "../../providers";
 
 import "./NavDrawer.scss";
 
-export const NavDrawer = ({ homeUrl = "/", docLinks }) => {
-  const { projectId } = useContext(CurrentProjectContext);
-  const { apps } = useContext(ApplicationsContext);
+export const NavDrawer = ({ docLinks }) => {
```

Comment:
Eventually, the child apps will be able to expose the list of sub-sections (such as Routers, Ensemblers, etc.) to be embedded into the navdrawer - correct?

Reply:
For the first milestone, I planned to keep it simple and to have all the side-nav items (and sub-items) to be configured in the parent app at run-time. The only responsibility of the child app is to expose the top-level routes of the application, which should include routes for the sub-sections too.

Later, this can be improved further by expecting child applications to also expose a remote service which would let the host app know what side-nav/project settings menu items must be rendered. Something like this:
https://adarshsomani64.medium.com/motivation-21e4f102677a > `Consuming common services`